src/app/components/routemap/RouteMap.tsx

```diff
@@ -0,0 +1,407 @@
+    ${tw`transition duration-150 ease-in-out mx-auto h-[14rem]`}
+    ${(props) => props.status === 'entered' || props.status === 'exit' ? tw`opacity-100` : tw`opacity-0`}
+`
+
+const RouteLine = styled.div`
+    ${tw`absolute transition duration-150 ease-in-out h-[3px] z-0`}
+`
+
+const SpecialStopsText = styled.p<{key:number, lang:string}>`
+    ${tw`absolute text-xs top-[-17px] left-[-15px] text-center w-10 font-bold`}
+    ${(props) => props.lang === 'ko' ? tw`tracking-tight` : tw`tracking-tighter text-[0.7rem]`}
+`
+
+export const RouteMap = (props: {
+    status:string,
+    tab: string
+}) => {
+    const timetable = useTimeTableContext().timetable
+
+    const { t, i18n } = useTranslation()
+    // dots
+    const [direct, setDirect] = useState<JSX.Element[]>([])
+    const [cycle, setCycle] = useState<JSX.Element[]>([])
+    const [yesulin, setYesulin] = useState<JSX.Element[]>([])
+    const [jungang, setJungang] = useState<JSX.Element[]>([])
+    // lines
+    const [dirLine, setDirLine] = useState<JSX.Element[]>([])
+    const [cycLine, setCycLine] = useState<JSX.Element[]>([])
+    const [yesLine, setYesLine] = useState<JSX.Element[]>([])
+    const [junLine, setJunLine] = useState<JSX.Element[]>([])
+    // dot refs
+    const refdir = useRef<Array<HTMLDivElement>>([])
+    const refcyc = useRef<Array<HTMLDivElement>>([])
+    const refyes = useRef<Array<HTMLDivElement>>([])
+    const refjun = useRef<Array<HTMLDivElement>>([])
+    // line refs
+    const linedir = useRef<Array<HTMLDivElement>>([])
+    const linecyc = useRef<Array<HTMLDivElement>>([])
+    const lineyes = useRef<Array<HTMLDivElement>>([])
+    const linejun = useRef<Array<HTMLDivElement>>([])
+    // langtest
+    const lang = useRef<Array<HTMLParagraphElement | null>>([])
+
+    const isPrevStop = useCallback((line: string, index: number) => {
+        if(props.tab === 'shuttlecoke_o'){
+            if (index === 0) return false
+            else return true
+        } else if (props.tab === 'shuttlecoke_i'){
+            if (line === 'direct') {
+                if (index < 3) return false
+                else return true
+            } else {
+                if (index < 4) return false
+                else return true
+            }
+        } else if (props.tab === 'subway'){
+            if (line === 'yesulin') return false
+            else {
+                if (index < 2) return false
+                else return true
+            }
+        } else if (props.tab === 'yesulin'){
+            if (line === 'direct' || line === 'jungang') return false
+            else {
+                if (index < 3) return false
+                else return true
+            }
+        } else if (props.tab === 'jungang'){ 
+            if (line === 'jungang') {
+                if (index < 3) return false
+                else return true
+            } else return false
+        } else {
+            // domitory is a terminal
+            return true
+        }
```

Comment:
This if-else statements can be simplified.

For example,

```typescript
if (index === 0) return false
else return true
```

Also, if it is simplified that way, I think there is no reason to use else-if statements since all if statements end with return statements. The code will never reach anyways if there are only if statements.

can be simplified as

```typescript
return index !== 0
```